ast/compile.go

```diff
@@ -3041,3 +3077,17 @@ func safetyErrorSlice(unsafe unsafeVars) (result Errors) {
 
 	return
 }
+
+func checkUnsafeBuiltinsInBody(unsafeBuiltinsMap map[string]bool, body Body) Errors {
```

Comment:
Nit: we could refactor this function to:

```golang
func checkUnsafeBuiltins(unsafeBuiltinsMap map[string]bool, node interface{}) Errors {
   // existing logic
}
```

This way it would accept either modules from the compiler or bodies from the query compiler.